backend/escalation_engine.py

```
@@ -248,25 +255,68 @@ def _escalate_grievance(self, grievance: Grievance, reason: EscalationReason,
             db.add(audit_log)
             db.commit()
 
+            # Update O(1) cache for next escalation audit (with max size limit to prevent memory leak)
+            with self._cache_lock:
+                if len(self._audit_last_hash_cache) >= 1000:
+                    self._audit_last_hash_cache.clear()
+                self._audit_last_hash_cache[grievance.id] = new_hash
+
             return True
 
         except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)